local/sdk-sriov/pkg/deviceplugin/server.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deviceplugin
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+)
+
+const (
+	resourceNamePrefix = "networkservicemesh.io/"
+	kubeletNotifyDelay = 30 * time.Second
+	vfioDir            = "/dev/vfio"
+)
+
+// Server is a SR-IOV forwarder device plugin server
+type Server interface {
+	// Start starts server with the given context
+	Start(ctx context.Context, manager Manager) error
+	// Stop stops server
+	Stop() error
+}
+
+type devicePluginServer struct {
+	resourceName  string
+	resourceCount int
+	hostBaseDir   string
+	hostPathEnv   string
+	ctx           context.Context
+	stop          context.CancelFunc
+}
+
+// NewServer creates a new SR-IOV forwarder device plugin server
+func NewServer(resourceName string, resourceCount int, hostBaseDir, hostPathEnv string) Server {
+	return &devicePluginServer{
+		resourceName:  resourceNamePrefix + resourceName,
+		resourceCount: resourceCount,
+		hostBaseDir:   hostBaseDir,
+		hostPathEnv:   hostPathEnv,
+	}
+}
+
+func (s *devicePluginServer) Start(parentCtx context.Context, manager Manager) error {
+	logFmt := "devicePluginServer(Start): %v"
+
+	s.ctx, s.stop = context.WithCancel(parentCtx)
+
+	socket, err := manager.StartDeviceServer(s.ctx, s)
+	if err != nil {
+		logrus.Errorf(logFmt, "error starting server")
+		return err
+	}
+
+	if err := manager.RegisterDeviceServer(s.ctx, &pluginapi.RegisterRequest{
+		Version:      pluginapi.Version,
+		Endpoint:     socket,
+		ResourceName: s.resourceName,
+	}); err != nil {
+		logrus.Errorf(logFmt, "error registering server")
+		return err
+	}
+
+	return nil
+}
+
+func (s *devicePluginServer) Stop() error {
```

Comment:
No need to stop in case of context approach.

Reply:
Fixed